### PR TITLE
DSET-3998 feat: export Logs resource info based on export_resource_info_on_event configuration

### PR DESCRIPTION
**Pull request by zdaratom-s1**
DSET-3998
Description: Introducing Logs resource info export to DataSet based on new export_resource_info_on_event configuration
Link to tracking Issue: https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/20660
Testing: verified via unit tests and also e2e with DataSet environment - no resource.* attributes gets to DataSet
Documentation: see exporter/datasetexporter/README.md

DSET-4007
Fix timestamp handling for LogRecord objects without a timestamp #2

### CODE REVIEW

**Review by martin-majlis-s1 — APPROVED** [review]
Great work @zdaratom . Thanks :)

File: exporter/datasetexporter/logs_exporter_test.go [hunk 1]
```diff
@@ -224,6 +221,102 @@ func TestBuildEventFromLog(t *testing.T) {
...
+		lr.ResourceLogs().At(0).Resource(),
+		lr.ResourceLogs().At(0).ScopeLogs().At(0).Scope(),
+		newDefaultLogsSettings(),
+	)
+
+	assert.Equal(t, expected, was)
+}
+
+func TestBuildEventFromLogEventWithoutTimestampWithOutObservedTimestampUseCurrentTimestamp(t *testing.T) {
+	// When LogRecord doesn't have timestampa and ObservedTimestamp set, current timestamp
```
Comment (atoulme) [inline exporter/datasetexporter/logs_exporter_test.go:290]:
```suggestion
	// When LogRecord doesn't have timestamp and ObservedTimestamp set, current timestamp
```

File: .chloggen/export-logs-resource-info-based-configuration [hunk 2]
```diff
@@ -0,0 +1,20 @@
+# Use this changelog template to create an entry for release notes.
+# If your change doesn't affect end users, such as a test fix or a tooling change,
+# you should instead start your pull request title with [chore] or use the "Skip Changelog" label.
+
+# One of 'breaking', 'deprecation', 'new_component', 'enhancement', 'bug_fix'
+change_type: 'enhancement', 'bug_fix'
```
Comment (atoulme) [inline .chloggen/export-logs-resource-info-based-configuration:6]:
```suggestion
change_type: 'enhancement'
```

File: exporter/datasetexporter/logs_exporter_test.go (hunk 1 above)
Comment (zdaratom) [reply exporter/datasetexporter/logs_exporter_test.go]: fixed

File: .chloggen/export-logs-resource-info-based-configuration (hunk 2 above)
Comment (zdaratom) [reply .chloggen/export-logs-resource-info-based-configuration]: fixed

**Review by atoulme — APPROVED** [review]

**Review by kovrus — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Pull request merged**